pkg/k8s/watchers/endpointsynchronizer.go

```diff
@@ -339,23 +339,37 @@ func updateCEPUID(scopedLog *logrus.Entry, e *endpoint.Endpoint, localCEP *ciliu
 	// If the Endpoint already owns the CEP (by holding the matching CEP UID reference) then we don't have to
 	// worry about other ownership checks.
 	//
-	// This will cover cases such as if the NodeIP changes (as with a reboot). In which case we can safely
-	// take ownership and overwrite the CEPs status.
+	// This will cover cases such as if the NodeIP changes (as with a reboot).
+	// In which case we can safely take ownership and overwrite the CEPs
+	// status. However if the cilium endpoints were previously checkpointed
+	// into tmpfs this check will fail and we will rely on the next check to
```

Comment:
nit: For future readers, we could make this comment more general: any situation where Cilium might lose its state after a restart would cause similar problems.